### DISCUSSION TITLE
[Feature Request] Manipulating story snaps post time

### DISCUSSION THREAD

**Discussion by Magna-Graecia** [rodit/SnapMod#105] [Ideas & Feature Suggestions]
I don't know if there's a way for you to make this happen, but we used to have the ability to manipulate the time a snap was posted to story. What I mean by this is say you were in London where the current time was 10 PM and you had a daytime snap that you wanted to upload, but you wanted to make it look like it was posted live earlier that day. What we used to be able to do was change our device date and time to the date/time we wanted, and then post to story as usual. So say you changed the time to 5 PM BST when it's currently actually 10 PM BST. This would cause the story snap to say "5 hours ago" even though you literally just posted it. What was funny too is that the story timer would be 5 hours short as well, so it would remain up for 19 hours from the point you posted as opposed to the full 24 hours since it thinks 5 hours have already passed.

However, this no longer works. Snapchat's server evidently no longer takes device time into account for live story snaps, so regardless of what your device time and zone is at the time of posting, it will show that it was just now posted (5 seconds ago, 10 seconds ago, etc). The only way I got this to work is by setting my device time to what I wanted, then sharing a photo/video from gallery to Snapchat, then tapping the "save" button in the bottom left so that it would be saved to the "snaps" section of memories, and posting from there. Snapchat uses the device time to create the timestamp for the snap in memories, so that timestamp will be compared to the actual current time in the given timezone when posting, and thus giving the "5 hours ago, etc" tag.

If I try doing this by going into a chat, opening the gallery drawer, long pressing on a snap to load it into the camera window, and tapping the "save" button in the bottom left to save it to memories, the snap gets saved to the "camera roll" section of memories instead of the "snaps" one, which causes "from camera roll" to show when posting. I could have sworn that it got saved to "snaps" before when doing this. I tried using the old-school workaround of hiding the snap so it got moved to "my eyes only", and then unhiding so it would be placed into the "snaps" section, but it would still say "from camera roll" when posting regardless. 

The only way this all works is by sharing the photo from gallery to Snapchat, and saving to memories from there. The problem with this is that it won't work if I try to share a video that's longer than 10 or 11 seconds. I get the "Sorry! Videos must be under 11 seconds to share" error. I suppose a workaround to all of this would be a bypass for that restriction the same as you've done for when loading videos from the gallery drawer to the camera window by long pressing on them, but this is assuming that the sharing from gallery to Snapchat, then saving to memories method continues to work.

So what do you think?

**Comment by rodit** [comment]
The last option seems most viable - bypassing the restriction globally shouldn't be too difficult.

**Comment by rodit** [comment]
See #106 and v1.6.1

**Comment by NMBlanco** [comment]
@Magna-Graecia wait I don't understand. So how exactly can you post pictures/videos from your gallery as live story now? That it doesn't show "from camera roll"?

**Comment by Magna-Graecia** [comment]
@rodit Awesome! You're the man! It's working perfectly. There's only one thing that people need to be aware of when using my method explained above for videos longer than 11 seconds now that we can share from gallery to Snapchat, then save to memories and go on to post from there to get whatever time stamp you want (2 hours ago, 5 hours ago, etc). For some reason when "Send Gallery as Snaps" is enabled in SnapMod settings, you can't send videos longer than 11 seconds from memories to story or even to people. If you disable this setting, then they will send fine, and using my method explained above you will get the timestamp you want + no "from camera roll" stamp. I wonder why this happens now. 

What's interesting is if you go into a chat, open the gallery drawer, and then swap the tab at the bottom from "Camera Roll" to "Memories", you can send whatever photo or video you want from there both by simply tapping on it to select it and sending (gets sent as 1 long snap this way), or by long pressing on it and loading it into the camera window, then sending (gets sent as multiple snaps this way). But if you go into "Memories" itself by swiping up on the camera screen and sending snaps from there, you can only send photos and videos that are 11 seconds or shorter, and they will get sent as live snaps while "Send Gallery as Snaps" is enabled. If you try sending any video that's longer than 11 seconds, nothing will happen. No error or anything. It just won't send, and there won't be any "sent" notifications or anything. Maybe something became disconnected there in this most recent update? And of course as stated, if you disable "Send Gallery as Snaps", then they will send fine from memories to story or people, but they won't go as live snaps to people in this case. They will go as chats, as is the default Snapchat way.

Either way, I'm able to work around it and accomplish my goal :) .. And the fact that we can now share videos longer than 11 seconds from gallery to Snapchat is HUGELY helpful in general, as it eliminates the need to endlessly scroll through the gallery drawer to find them.

@tarek133 If you just want to post photos/videos to story without the "from camera roll" tag, just open a chat, open the gallery drawer, and press and hold on the photo/video you want to send to load it into the camera window. From there, tap at the bottom of the screen where it shows who it's being sent to. By default it will be filled with the person whose chat window you were in. Simply deselect that person and select "My Story" and send. You will see that it doesn't say "from camera roll". You can also choose to have both "My Story" and multiple other people selected if you want to send to multiple places at once, same as normal sending.

If you want to post photos/videos to story without the "from camera roll" tag AND with a specific timestamp (like 5 hours ago, etc), then you will need to follow my method explained in my posts above.

**Comment by Magna-Graecia** [comment]
Just wanted to update with some info. I tried my trick on an unrooted/unmodded device where I share a photo/video from gallery to Snapchat, save it to memories, then post it to "My Story" from there to get a custom timestamp. As suspected, it works, and I don't get any "from camera roll" tags. This makes sense though of course, since the trick doesn't really have anything to do with SnapMod. The benefit with SnapMod when using this trick is that we can now share videos longer than 11 seconds from gallery to Snapchat to do it with. On an unrooted/unmodded device, or just one that's not using SnapMod, we will get the "Sorry! Videos must be under 11 seconds to share" error.

I also tried sending an 11+ second long video from memories to both story and to a user (my other Snapchat account). It works of course. It gets posted to story with either no tag (if it was saved to memories less than 24 hours ago), or with a "1 day ago, 1 week ago, 1 month ago from memories etc" tag. When sending to a user, it gets sent as a chat.

As mentioned in my post above, if you try to send a photo or video under 11 seconds long on a device with SnapMod and "Send Gallery as Snaps" enabled, they will go as live snaps. If you disable "Send Gallery as Snaps", they will go as chats. This is normal. However, if you try to send an 11+ second long video while "Send Gallery as Snaps" is enabled, it won't get sent, and no error notifications will show. Simply nothing will happen. If you try to send an 11+ second long video while "Send Gallery as Snaps" is disabled, then it will get sent as a chat. So what I suspect is happening is basically the global video length bypass is not being applied to any snaps sent from within memories. It applies if you're using the gallery drawer within a chat and switch the tab at the bottom from "Camera Roll" to "Memories", but not when you're using the "Memories" window itself. This is why videos longer than 11 seconds aren't being sent as live snaps from there while "Send Gallery as Snaps" is enabled, but photos and videos shorter than 11 seconds are. It also explains why there is no error notification since you're not supposed to be able to send photos/videos as live snaps from memories to begin with, so there never was a an error notification made for when someone tries to do it with an 11+ second long snap haha.

I guess the solution would be if you could apply the global video length bypass to snaps sent from memories as well. This way we wouldn't need to disable "Send Gallery as Snaps" when using my trick explained above to post an 11+ second long video to "My Story" to get a custom timestamp. Not that it's a big deal, since we can already do it right now with some workaround. But perhaps if it's easy for you to apply it.
